pinot-common/src/main/java/com/linkedin/pinot/common/utils/DataTableJavaSerDe.java

```diff
@@ -36,34 +35,19 @@
 public class DataTableJavaSerDe implements DataTableSerDe {
   private static final Logger LOGGER = LoggerFactory.getLogger(DataTableJavaSerDe.class);
 
-  /**
-   * @param object Object to serialize
-   * @return Serialized byte-array of the object
-   */
   @Override
   public byte[] serialize(Object object) {
     return serializeJavaObject(object);
   }
 
-  /**
-   * @param bytes Serialized byte-array
-   * @param dataType Enum type of the object
-   * @param <T> Type of the object
-   * @return De-serialized object
-   */
   @Override
-  public <T extends Serializable> T deserialize(byte[] bytes, DataType dataType) {
+  public <T> T deserialize(byte[] bytes, DataType dataType) {
```

Comment:
It's an override. Because in customized serDe we don't enforce it to be Serializable. I will clean up the JavaSerDe in another PR, and all objects should be ser/de by customized one.